src/aiidalab_qe/app/result/utils/download_data.py

```diff
@@ -0,0 +1,160 @@
+import base64
+import pathlib
+import tempfile
+from threading import Thread
+
+import ipywidgets as ipw
+
+
+class DownloadDataWidget(ipw.VBox):
+    def __init__(self, workchain_node):
+        self.download_archive_button = ipw.Button(
+            description="Download AiiDA archive.aiida data",
+            icon="download",
+            button_style="primary",
+            disabled=False,
+            tooltip="Download the AiiDA archive of the simulation, ready to be shared or imported into another AiiDA profile",
+            layout=ipw.Layout(width="auto"),
+        )
+        self.download_archive_button.on_click(self._download_data_thread)
+
+        self.download_raw_button = ipw.Button(
+            description="Download AiiDA raw data (zip format)",
+            icon="download",
+            button_style="primary",
+            disabled=False,
+            tooltip="Download the raw data of the simulation, organized in intuitive directory paths.",
+            layout=ipw.Layout(width="auto"),
+        )
+        self.download_raw_button.on_click(self._download_data_thread)
+
+        self.node = workchain_node
+
+        super().__init__(
+            children=[
+                ipw.HTML(
+                    "<b>Download the data</b>"
+                    "<br>It is possible to download raw data (i.e. input and output files) and/or "
+                    "<br>the AiiDA archive (ready to be shared or imported into another AiiDA profile)"
+                ),
+                ipw.HBox(
+                    children=[self.download_raw_button],
+                    layout=ipw.Layout(width="500px"),  # Set the desired width here
+                ),
+                ipw.HBox(
+                    children=[self.download_archive_button],
+                    layout=ipw.Layout(width="500px"),  # Set the desired width here
+                ),
+            ],
+        )
+
+    def _download_data_thread(self, button_instance):
+        thread = Thread(target=lambda: self._download_data(button_instance))
+        thread.start()
+
+    def _download_data(self, button_instance):
+        """
+        This method handles the download process when a download button is clicked.
+        It updates the button's description to indicate that the download is in progress,
+        determines whether to download the archive or raw data based on the button's description,
+        generates the appropriate bitstream from the specified node, initiates the download
+        with a filename based on the node's primary key, and then resets the button description
+        to its original state.
+
+        Args:
+            button_instance (ipywidgets.Button): The button instance that was clicked.
+        """
+
+        if "archive" in button_instance.description:
+            what = "archive"
+            filename = f"export_{self.node.pk}.aiida"
```

Comment:
It's would be good to add `qeapp` to the name of the file, e.g.,
```suggestion
            filename = f"export_qeapp_calculation_pk_{self.node.pk}.aiida"
```
Same for the raw data